pages/agent/plugins.md.erb

```diff
@@ -0,0 +1,124 @@
+
+# Buildkite Agent Plugins
+
+__Plugins are only available in the beta 3.0 version of the agent__
+
+Plugins provide a way to extend the functionality that Buildkite natively offers. Essentially, they are neatly packaged collections of [Hooks](hooks) that run as part of your builds. Buildkite offers a [collection of useful plugins](https://github.com/buildkite-plugins) for things like Docker integration, Github Releases and interaction with Amazon Web Services and Google Cloud. 
+
+<%= toc %>
+
+## Using a plugin
+
+Plugins are only available for [pipelines] expressed in yaml, so all of these examples are going to be using that syntax.  
+
+Plugins are defined per-step and are automatically fetched from github based on the name of the project. For instance, checkout our clown-detection plugin which scans your code for pesky clowns:
+
+```yml
+steps:
+  - plugins:
+      detect-clowns:
+        emojis: 🤡
+```
+
+This example will install [buildkite-plugins/detect-clowns-buildkite-plugin](https://github.com/buildkite-plugins/detect-clowns-buildkite-plugin) and run it for this build step.
+
+You could just as easily write your own plugin on GitHub at `myusername/blah-buildkite-plugin` and it would be available as `myusername/blah` in your pipeline.yml.
```

Comment:
Keep an eye out for using the word 'easy'! We don't want to alienate people who find this difficult 😊